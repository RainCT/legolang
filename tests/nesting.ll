//bool cond1

def a():
    def b():
        print "in b"
        c()
        def c():
            print "in c"
            if true: //cond1:
                int expectedValue = 5;
                def d() returns int:
                    print "in d (1)"
                    e()
                    return 5
                if d() != expectedValue:
                    print "FAIL"
            else:
                def d():
                    print "in d (2)"
                    e()
                d()
            print "out d"
        print "out c"
    def e():
        print "here is e"
    print "in a"
    b()
    print "out b"
    print "out a"

//cond1 = true
a()

// FIXME: implementar globals, i moure aix√≤ a un altre test nou!
/*print ""
cond1 = false
a()*/
