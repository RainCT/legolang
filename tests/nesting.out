in a
in b
in c
in d (1)
here is e
out d
out c
out b
out a
